docs/source/sparse.rst

```diff
@@ -363,6 +363,78 @@ assumption that the fill value is negative infinity.
 
 .. See https://github.com/Quansight-Labs/rfcs/tree/pearu/rfc-fill-value/RFC-0004-sparse-fill-value for a new API
 
+.. _sparse-csr-docs:
+
+Sparse CSR Tensor
++++++++++++++++++
+
+The CSR (Compressed Sparse Row) sparse tensor format implements the CSR format
+for storage of 2 dimensional tensors. Although there is no support for N-dimensional
+tensors, the primary advantage over the COO format is better use of storage and
+much faster computation operations such as sparse matrix-vector multiplication
+using MKL and MAGMA backends. CUDA support does not exist as of now.
+
+A CSR sparse tensor consists of three 1-D tensors: ``crow_indices``, ``col_indices``
+and ``values``:
+
+  - The ``crow_indices`` tensor consists of compressed row indices. This is a 1-D tensor
+    of size ``size[0] + 1``. The last element is the number of non-zeros.
+  - The ``col_indices`` tensor contains the column indices of each value. This is a 1-D
+    tensor of size ``nnz``.
+  - The ``values`` tensor  contains the values of the CSR tensor. This is a 1-D tensor
+    of size ``nnz``.
+
+.. note::
+
+   The index tensors ``crow_indices`` and ``col_indices`` should have element type either
+   ``torch.int64`` (default) or ``torch.int32``. If you want to use MKL-enabled matrix
+   operations, use ``torch.int32``. This is as a result of the default linking of pytorch
+   being with MKL LP64, which uses 32 bit integer indexing.
```

Comment:
So... why don't we default to int32 indices?

Reply:
COO defaults to int64 and MKL operations are the only ones that need 32 bit indexing. What ever does not use MKL can use 64 bit indices. So if the user wants to use MKL they can use 32 bit indices.